api/base/serializers.py

```diff
@@ -120,13 +118,15 @@ def to_representation(self, value):
         the link is represented as a links object with 'href' and 'meta' members.
         """
         url = super(JSONAPIHyperlinkedIdentityField, self).to_representation(value)
+        if not url:
+            raise SkipField
 
         meta = {}
         for key in self.meta or {}:
             if key == 'count':
```

Comment:
After further discussion with @sloria, this was identified as a bug to be fixed separately:
https://openscience.atlassian.net/browse/OSF-4832